utils/bt_plugins/comparator.rb

```diff
@@ -29,6 +30,9 @@ def initialize_method(self_component, _configuration, _params, _data)
   end
 
   def verify_messages
+    if @stack_messages.values.map { |data| data.length }.to_set.length > 1
```

Comment:
Good catch! 

If they are not the same size, `transpose`should fail with an error. Then it possible to `rescue` to print a better error message if it was not clear enough.

PS: You can do `@stack_messages.values.uniq{ |data| data.length }.length  > 1 ` (so no need to rely on set :)